### ISSUE TITLE
AppID change - How to ? Help Wanted

### ISSUE THREAD

**Issue by MopheusDG** [NativeScript/nativescript-cli#3939]
Hi everyone, I'm needing some help to try to clarify as following the procedure to change the AppID in the NS documentation for publishing is not working as expected, and after reading so many posts about the issue, I'm still a little confused (Sorry).

The NS page says:
[https://www.nativescript.org/blog/steps-to-publish-your-nativescript-app-to-the-app-stores#step-3](https://www.nativescript.org/blog/steps-to-publish-your-nativescript-app-to-the-app-stores#step-3)

> Open your app’s root package.json file and find the "nativescript" key. Make sure the "id" attribute contains the value you’d like to use.
> NOTE: If you’re switching your application id in your package.json file, you might also need to change the value in your app/App_Resources/Android/app.gradle file (find the applicationId key), for the change to take effect on Android.

What I found in the forum/issues/etc is that I should change this values:

**- package.json** - "nativescript": { "id": "com.example.newapp" ...
**- app/App_Resources/Android/app.gradle** - ... applicationId = "com.example.newapp" ...
**- app/App_Resources/Android/settings.json** - {"appId":"com.example.newapp"...

So, I want to be sure, what exactly needs to be change in NS 4.2.x to change my AppID for both, ANDROID and IOS ?

Can someone help me please ?

And by the way, do I need to manually change anything in the PLIST for IOS too ?

**Comment by NickIliev** [comment]
To change the app ID on Android use the above three steps. On the other hand on iOS changing the `nativescript.id` in your `package.json` is enough and no further changes are required. You can also change the appID via a CLI flag passed with your build command - see [this documentation section](https://docs.nativescript.org/tooling/publishing/publishing-ios-apps#bundle-id) explaining about the bundle ID and the application name change on iOS

**Comment by MopheusDG** [comment]
Thanks Nick. Seems to be working fine with the 3 changes as you said.

**Comment by aosi87** [comment]
This is still a must on NS6, dunno why it does update it on android sometimes, the new method is not 100% accurate, cheers.

**Comment by NathanaelA** [comment]
@aosi87 - Isacc, not understanding the issue you are stating; if you change the package.json key, then you will want to delete the platforms folder so that when `tns` reinstalls the platforms it will use your new name.  Otherwise you will be out of sync, and things will work weirdly.

**Comment by aosi87** [comment]
I did update the package.json, deleted the package-lock.json, platforms folder, hooks, and used tns prepare android command, after that, I checked App_Resources/Android/src/main/settings.json and it has the old value, did this twice, and it never gets updated. 

I created the project using Nativescript companion app sidekick. after the notice of dumping cloud build capability, I switched to CLI only, and start noticing this behavior.

Current stack:
```
nativescript. Your current version is 6.5.0 and the latest available version is 6.7.4.
tns-core-modules. Your current version is 6.5.1 and the latest available version is 6.5.6.
tns-android. Your current version is 6.5.0 and the latest available version is 6.5.1.
tns-ios. Your current version is 6.5.0 and the latest available version is 6.5.1.
```
After updating app.gradle manually the platform temp settings worked.
So in my experience, this still required sometimes in every other prepare command

**Comment by NathanaelA** [comment]
. `App_Resources/Android/src/main/settings.json` This is not a file the CLI handles -- so you can actually delete it if you are no longer using Sidekick

This is what a new project looks like;
![image](https://user-images.githubusercontent.com/850871/84929527-8c60ef00-b095-11ea-8641-8cce64dada97.png)

In addition on android your `AndroidManifest.xml` should look like:
![image](https://user-images.githubusercontent.com/850871/84929716-d518a800-b095-11ea-9014-ad3ef3274dac.png)
The `	package="__PACKAGE__" ` is rewritten during building to be what you have in your package.json fole..
